src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -326,6 +333,59 @@ func (r *ClusterReconciler) setLicense(
 	return nil
 }
 
+func (r *ClusterReconciler) setPodNodeIDAnnotation(
```

Comment:
just curious: will statefulset overwrite this in the next rollout?

Reply:
No, but even if it did, it'll get annotated again if the annotation is not there.